src/docfx/lib/schema/JsonSchemaProvider.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.IO;
+
+namespace Microsoft.Docs.Build
+{
+    internal class JsonSchemaProvider
+    {
+        private ConcurrentDictionary<string, JsonSchema> _jsonSchemas = new ConcurrentDictionary<string, JsonSchema>();
+
+        // TODO: get schema from template
+        public JsonSchema GetJsonSchema(Schema schema)
```

Comment:
move this method to `TemplateEngine`, there is no need to create a new type and pass it down

Reply:
context.Template could be null in our test pipeline, so do you mean put a static method in Template?